.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,38 @@
+name: "Prepare release"
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - 'v*'
+
+env:
+  CI: true
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    permissions:
+      contents: write  # for softprops/action-gh-release to create GitHub release
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - run: git fetch --tags -f
+
+      - name: Resolve version
+        id: vars
+        run: |
+          echo "TAG_NAME=$(git describe --tags --abbrev=0)" >> $GITHUB_ENV
+
+      - name: Create Release
+        uses: softprops/action-gh-release@v1
+        with:
+          draft: true
```

Comment:
Depending on what you want you could opt for automatic publishing. For the design library I find it very important to have good release notes that I want to write manually, but if this is a minor package that might not be needed. The `generate_release_notes` option below ensure that the release will contain a list of all merged PR's. That does mean any manual commits to `main` are not included, so if that is the workflow here then manual release notes are still needed.